codec-memcache/src/main/java/io/netty/handler/codec/memcache/binary/DefaultFullBinaryMemcacheResponse.java

```diff
@@ -113,6 +113,36 @@ public FullBinaryMemcacheResponse duplicate() {
         if (extras != null) {
             extras = extras.duplicate();
         }
-        return new DefaultFullBinaryMemcacheResponse(key, extras, content().duplicate());
+        return copy(key, extras, content().duplicate());
+    }
+
+    @Override
+    public FullBinaryMemcacheResponse retainedDuplicate() {
+        ByteBuf key = key();
```

Comment:
`return replace(key, extras, content().retainedDuplicate());` ... assuming `replace` should also be calling `retainedDuplicate()` on `key` and `extras`.